sw/device/lib/dif/dif_entropy_src.c

```diff
@@ -533,17 +534,23 @@ dif_result_t dif_entropy_src_observe_fifo_write(
     return kDifError;
   }
 
-  // Check if the FIFO is full.
-  if (mmio_region_read32(entropy_src->base_addr,
-                         ENTROPY_SRC_FW_OV_WR_FIFO_FULL_REG_OFFSET)) {
-    return kDifIpFifoFull;
-  }
-
+  // Check if the FIFO is full before writing each word.
   for (size_t i = 0; i < len; ++i) {
+    if (mmio_region_read32(entropy_src->base_addr,
+                           ENTROPY_SRC_FW_OV_WR_FIFO_FULL_REG_OFFSET)) {
+      if (written) {
```

Comment:
@engdoreis 
couldn't we just always assign the value of i to written here..?

Reply:
This depends on the intention for the API.
In this case the new argument `written` is optional which means that the caller may not use it and just pass a `NULL`, so before assigning we need to check whether is `NULL`.  

I personally like this approach.

Reply:
o crap it's for checking null, i missed that. 
thanks.

Reply:
I thought about updating it after every write, but then I perhaps over-optimized and just set it to update once on exit.   You do need to check whether the space is non-null.  (Since I'm sticking my nose in an unusual OT-role for myself, I also re-read the C-styleguide which said somewhere that NULL values to output parameters should always expect NULL as a possible value).

Reply:
Is there a more standard way to check for NULL?